src/Model/State.re

```diff
@@ -61,6 +61,7 @@ type t = {
   terminals: Feature_Terminal.t,
   layout: Feature_Layout.t(int),
   fileExplorer: FileExplorer.t,
+  window: Revery.Window.t,
```

Comment:
I'm wondering though if we actually want `window` in the state, because it's not serializable. One of the cool things about maintaining all application state in a single data struture is that you can fully re-create an application's state just by serializing it in one place, then move and deserialize it in another. You could for example attach it to a bug report and have the developer re-create the exact conditions of a tricky bug on their own computer. But that doesn't work if the data structure contains unserializable values like functions or opaque external objects like `window` here.

This certainly isn't the only value that is currently unserializable in the state though, so it wouldn't make much difference in itself, but it would be nice if we tried to move toward rather than away from being serializable.

One way we could do this is to maintain the invariant that a workspace has a single window, and keep the mapping from workspace to window outside the state. It could just be a lookup table from workspace path to window instance, for example, so that in order to get the current window of a workspace we'd just call `windowFor(workspacePath)`. Another benefit of this is that it would be a step towards supporting multiple windows. At least as long as we stick with the workspace-window invariant.

Not sure if it's worth the bother just for this though.

Reply:
Another benefit of serializable values is that they're also comparable structurally, which is important for some performance optimizations. By checking for structural equality we can avoid re-renders as there's no point in re-rendering if nothing's changes. But if we try comparing values that aren't comparable, we'll crash. Which isn't good.

Reply:
Hey @glennsl! Thanks for the warm welcome! I thought putting the window in the state might be a controversial idea (I said as much in my original PR haha). I am by no means tied to that idea. I will look at the setTitle example you provided and see if I can modify the maximize code to fit that.

Thanks again!